contracts/external/dao-voting-incentives/src/msg.rs

```diff
@@ -1,42 +1,56 @@
 use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::Uint128;
+use cw20::Cw20ReceiveMsg;
+use cw_denom::{CheckedDenom, UncheckedDenom};
+use cw_ownable::cw_ownable_query;
+use cw_utils::Expiration;
 use dao_hooks::vote::VoteHookMsg;
-
-use crate::state::VotingIncentives;
+use dao_interface::state::Config;
 
 #[cw_serde]
 pub struct InstantiateMsg {
-    /// DAO address
-    pub dao: String,
-    /// Rewards to pay out for voting.
-    pub voting_incentives: VotingIncentives,
+    /// The contract's owner using cw-ownable
+    pub owner: String,
+    /// The denom to distribute
+    pub denom: UncheckedDenom,
+    /// The expiration of the voting incentives
+    pub expiration: Expiration,
 }
 
 #[cw_serde]
 pub enum ExecuteMsg {
     /// Fires when a new vote is cast.
     VoteHook(VoteHookMsg),
-    /// Claim rewards.
+    /// Claim rewards
     Claim {},
+    /// Expire the voting incentives period
+    Expire {},
+    UpdateOwnership(cw_ownable::Action),
+    Receive(Cw20ReceiveMsg),
 }
 
+#[cw_ownable_query]
 #[cw_serde]
 #[derive(QueryResponses)]
 pub enum QueryMsg {
-    /// Returns the config.
-    #[returns(ConfigResponse)]
+    /// Returns the config
+    #[returns(Config)]
     Config {},
-    /// Returns the rewards for the given address.
-    #[returns(cosmwasm_std::Uint128)]
+    /// Returns the claimable rewards for the given address.
+    #[returns(RewardResponse)]
     Rewards { address: String },
+    /// Returns the expected rewards for the given address
+    #[returns(RewardResponse)]
+    ExpectedRewards { address: String },
```

Comment:
What is the difference between these two queries? They seem the same to me... maybe I am missing something.

Reply:
Rewards is meant to be used after expiration, because the balance is set in config. The value refers to the actual claimable amount.

ExpectedRewards is meant to be used before the voting incentives are expired, because the vote counts and balance can still change.

Reply:
Hmmmm, a little confusing. Maybe others have thoughts, but perhaps we could combine these and use a custom response type to indicate whether incentives are expired and the amount that is expected vs claimable?

Open to other ideas.

Reply:
Yeah, I think the combination would be a good idea, and I could add an `is_claimable` flag to the response.